sdk/storage/Azure.Storage.Files.Shares/src/ShareFileClient.cs

```diff
@@ -683,6 +687,460 @@ private void SetNameFieldsIfNull()
+        /// </param>
+        /// <param name="metadata">
+        /// Optional custom metadata to set for the file.
+        /// </param>
+        /// <param name="smbProperties">
+        /// Optional SMB properties to set for the file.
+        /// </param>
+        /// <param name="filePermission">
+        /// Optional file permission to set on the file.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="ShareFileRequestConditions"/> to add conditions
+        /// on creating the file.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{StorageFileInfo}"/> describing the
+        /// state of the file, or <c>null</c> if the file already exists.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<ShareFileInfo>> CreateIfNotExistsInternal(
+            long maxSize,
+            ShareFileHttpHeaders httpHeaders,
+            Metadata metadata,
+            FileSmbProperties smbProperties,
+            string filePermission,
+            ShareFileRequestConditions conditions,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            using (Pipeline.BeginLoggingScope(nameof(ShareFileClient)))
+            {
+                Pipeline.LogMethodEnter(
+                    nameof(ShareFileClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}");
+                try
+                {
+                    string operationName = $"{nameof(ShareFileClient)}.{nameof(CreateIfNotExists)}";
+
+                    Response<bool> existsResponse = await ExistsInternal(
+                        async,
+                        cancellationToken,
+                        operationName).ConfigureAwait(false);
+
+                    if (existsResponse.Value)
+                    {
+                        return null;
+                    }
+                    else
+                    {
+                        return await CreateInternal(
+                            maxSize,
+                            httpHeaders,
+                            metadata,
+                            smbProperties,
+                            filePermission,
+                            conditions,
+                            async,
+                            cancellationToken,
+                            operationName)
+                            .ConfigureAwait(false);
+                    }
```

Comment:
This will cause race conditions.  It would be better to try the Create call and ignore errors if it already exists like you did for the directory client.

Reply:
[https://docs.microsoft.com/en-us/rest/api/storageservices/create-file](https://docs.microsoft.com/en-us/rest/api/storageservices/create-file)

I can't call Create an ignore errors, Create File overwrites existing files, and there are no conditional request headers.  I think this might be the only way to do `CreateIfNotExists` for Files.

Reply:
Oh...  yeah, that's kind of scary.  We don't want to give people APIs they can't be successful with so I think we should only add `Exists` and `DeleteIfExists` for now for File Shares.

Reply:
Removed `CreateIfNotExists` from File Share.